src/fuzzer/seeds.rs

```diff
@@ -2,23 +2,25 @@
 //! handshake or an execution which crashes OpenSSL.
 #![allow(dead_code)]
 
-use rustls::internal::msgs::enums::{HandshakeType, Compression};
+use rustls::internal::msgs::enums::{Compression, HandshakeType};
 
 use crate::agent::{AgentDescriptor, TLSVersion};
+use crate::concretize::PUTType;
+use crate::concretize::PUTType::OpenSSL;
 use crate::term;
+use crate::tls::fn_impl::*;
 use crate::trace::TlsMessageType;
 use crate::trace::TlsMessageType::Handshake;
-use crate::tls::fn_impl::*;
 use crate::{
     agent::AgentName,
     term::Term,
     trace::{Action, InputAction, OutputAction, Step, Trace},
 };
-use rustls::{ProtocolVersion, CipherSuite};
-use rustls::msgs::handshake::{SessionID, Random};
 use rustls::internal::msgs::handshake::ServerExtension;
+use rustls::msgs::handshake::{Random, SessionID};
+use rustls::{CipherSuite, ProtocolVersion};
 
-pub fn seed_successful(client: AgentName, server: AgentName) -> Trace {
+pub fn seed_successful(client: AgentName, server: AgentName, put_type: PUTType) -> Trace {
```

Comment:
Just a quick idea: We could make `AgentName` generic over a PUT. That way we could infer which PUT to spawn based on the AgentName.

Reply:
The serializability of traces would be a problem then though.

Reply:
agent::AgentDescriptor does that, it now has a field `pub put_type: PUTType`. PUTType is just an enum so no problem for serializability. It just keeps track of which PUT will have to be used for executing/animating that agent.